public/app/features/manage-dashboards/DashboardImportPage.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { FormEvent, PureComponent } from 'react';
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { css } from 'emotion';
+import { AppEvents, NavModel } from '@grafana/data';
+import { Forms, stylesFactory } from '@grafana/ui';
+import Page from 'app/core/components/Page/Page';
+import { ImportDashboardOverview } from './components/ImportDashboardOverview';
+import { DashboardFileUpload } from './components/DashboardFileUpload';
+import { fetchGcomDashboard, importDashboardJson } from './state/actions';
+import appEvents from 'app/core/app_events';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { StoreState } from 'app/types';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  navModel: NavModel;
+  isLoaded: boolean;
+}
+
+interface DispatchProps {
+  fetchGcomDashboard: typeof fetchGcomDashboard;
+  importDashboardJson: typeof importDashboardJson;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+interface State {
+  gcomDashboard: string;
+  dashboardJson: string;
+  dashboardFile: any;
+  gcomError: string;
+}
+
+const importStyles = stylesFactory(() => {
+  return {
+    option: css`
+      margin-bottom: 32px;
+    `,
+  };
+});
+
+class DashboardImportUnConnected extends PureComponent<Props, State> {
+  state: State = {
+    gcomDashboard: '',
+    dashboardJson: '',
+    dashboardFile: '',
+    gcomError: '',
+  };
+
+  onGcomDashboardChange = (event: FormEvent<HTMLInputElement>) => {
+    this.setState({ gcomDashboard: event.currentTarget.value });
+  };
+
+  onDashboardJsonChange = (event: FormEvent<HTMLTextAreaElement>) => {
+    this.setState({ dashboardJson: event.currentTarget.value });
+  };
+
+  onFileUpload = (event: FormEvent<HTMLInputElement>) => {
+    const { importDashboardJson } = this.props;
+    const file = event.currentTarget.files[0];
+
+    const reader = new FileReader();
+    const readerOnLoad = () => {
+      return (e: any) => {
+        let dashboard: any;
+        try {
+          dashboard = JSON.parse(e.target.result);
+        } catch (error) {
+          appEvents.emit(AppEvents.alertError, ['Import failed', 'JSON -> JS Serialization failed: ' + error.message]);
+          return;
+        }
+        importDashboardJson(dashboard);
+      };
+    };
+    reader.onload = readerOnLoad();
+    reader.readAsText(file);
+  };
+
+  onDashboardJsonLoad = () => {
+    const { dashboardJson } = this.state;
+    const { importDashboardJson } = this.props;
+
+    if (!dashboardJson) {
+      return;
+    }
+    let dashboard: any;
+    try {
+      dashboard = JSON.parse(dashboardJson);
+      importDashboardJson(dashboard);
+    } catch (error) {
+      appEvents.emit(AppEvents.alertError, ['Import failed', 'JSON -> JS Serialization failed: ' + error.message]);
+      return;
+    }
+  };
```

Comment:
Same as with the load from gcom -> the textarea for json paste could be a separate form validated in a async way, so the error could be shown inline rather than with the App notifications, wdyt?